### PR TITLE
Range/Slice AST

### PR DESCRIPTION
**Pull request by helixbass**
@GeoffreyBooth AST for `Range`/`Slice`

Based on `call-ast`, [here](https://github.com/helixbass/copheescript/compare/call-ast...range-ast) is just the diff against that branch

### CODE REVIEW

File: src/nodes.coffee [hunk 1]
```diff
@@ -1683,6 +1697,12 @@ exports.Range = class Range extends Base
     args   = ', arguments' if hasArgs(@from) or hasArgs(@to)
     [@makeCode "(function() {#{pre}\n#{idt}for (#{body})#{post}}).apply(this#{args ? ''})"]
 
+  astProperties: -> {
```
Comment (helixbass) [inline src/nodes.coffee]: `Range` is a Coffeescript-specific node type, but its AST representation seems pretty clear-cut

File: src/nodes.coffee [hunk 2]
```diff
@@ -1719,6 +1739,9 @@ exports.Slice = class Slice extends Base
           "+#{fragmentsToText compiled} + 1 || 9e9"
     [@makeCode ".slice(#{ fragmentsToText fromCompiled }#{ toStr or '' })"]
 
+  ast: ->
```
Comment (helixbass) [inline src/nodes.coffee]: `Slice` just forwards AST generation to `Range`

File: test/abstract_syntax_tree.coffee [hunk 3]
```diff
@@ -561,90 +561,121 @@ test "AST as expected for Index node", ->
...
-#   testExpression 'x[...]',
-#     properties: [
-#       range:
-#         type: 'Range'
-#         exclusive: yes
-#         equals: ''
-#         from: undefined
-#         to: undefined
-#     ]
+  # testExpression 'for x in [42...43] then',
```
Comment (GeoffreyBooth) [inline test/abstract_syntax_tree.coffee]: I’m assuming these tests are commented out because we don’t yet support AST output for `for` or `in`?

**Comment by GeoffreyBooth** [comment]
Just one question, otherwise this looks good.

File: test/abstract_syntax_tree.coffee (hunk 3 above)
Comment (helixbass) [reply test/abstract_syntax_tree.coffee]: Yup exactly so just included tests for "standalone" ranges and slices for now

**Pull request merged**